librz/util/iterator.c

```diff
@@ -64,6 +64,28 @@ RZ_API RZ_BORROW void *rz_iterator_next(RZ_NONNULL RZ_BORROW RzIterator *it) {
 	return it->cur;
 }
 
+/**
+ * \brief Fetches the last element with the RzIterator
+ *
+ * This function retrieves the last element in the sequence for a given RzIterator.
+ * Before fetching the last element, it frees the current element using the
+ * RzIterator's `free` function if it's not NULL.
+ *
+ * \param it A pointer to the RzIterator object.
+ *
+ * \return void* Pointer to the element gotten as the last object in the sequence or NULL if operation failed.
+ */
+RZ_API RZ_BORROW void *rz_iterator_last(RZ_NONNULL RZ_BORROW RzIterator *it) {
+	rz_return_val_if_fail(it && it->next, NULL);
+	void *elem = NULL;
+	void *tail = rz_iterator_next(it);
+	while (tail) {
+		elem = tail;
+		tail = rz_iterator_next(it);
```

Comment:
I'm wondering if it's possible to free elem in rz_iterator_next.

Reply:
Good point, I removed the function call and used `->next()` method instead.